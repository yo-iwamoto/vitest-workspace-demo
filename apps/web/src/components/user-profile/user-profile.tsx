type Props = {
  userName: string;
};

export function UserProfile({ userName }: Props) {
  return (
    <div>
      <p>{userName}</p>
    </div>
  );
}
