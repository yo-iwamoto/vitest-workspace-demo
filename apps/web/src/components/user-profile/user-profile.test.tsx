import { UserProfile } from './user-profile';
import { render } from '@testing-library/react';

describe('UserProfile', () => {
  it('snapshot unchanged', () => {
    const { asFragment } = render(<UserProfile userName='John Doe' />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('ユーザー名を表示すること', () => {
    const { getByText } = render(<UserProfile userName='John Doe' />);

    expect(getByText('John Doe')).toBeDefined();
  });
});
