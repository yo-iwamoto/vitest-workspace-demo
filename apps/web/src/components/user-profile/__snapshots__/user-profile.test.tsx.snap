// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserProfile > snapshot unchanged 1`] = `
<DocumentFragment>
  <div>
    <p>
      John Doe
    </p>
  </div>
</DocumentFragment>
`;
